core/models.py

```diff
@@ -528,10 +534,12 @@ def is_active(self):
         return True
 
     def has_perm(self, perm, obj=None):
-        i_user = self.i_user if obj is None else obj.i_user
+        self.has_perms(self, perm, obj=None)
```

Comment:
Don't overwrite this method in this way, has_perm is a django function and from what I see has_perms is using has perm to do the check. Therefore there will be recursion exception. You should do the check for superadmin and then return either true or super(User, self).has_perm